src/lib/dnssd/Resolver.cpp

```diff
@@ -0,0 +1,41 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "Resolver.h"
+
+namespace chip {
+namespace Dnssd {
+
+Resolver * Resolver::sInstance = nullptr;
+
+Resolver & Resolver::Instance()
+{
+    if (sInstance == nullptr)
+    {
+        sInstance = &GetDefaultResolver();
```

Comment:
I might me missing something, but if you have both minimal and platform, doesn't both GetDefaultResolver conflict?

Same for advertiser.

Reply:
The implementations can be built independently using new `chip_mdns_minimal` and `chip_mdns_platform` GN args but there's a single default resolver/advertiser selected with the existing `chip_mdns` flag.

Reply:
It won't, because there are two kinds of flags deciding about it. `chip_mdns_minimal` and `chip_mdns_platform` decide about including the files related to these implementations to the compilation, so we can add both. On the other hand about the `GetDefaultResolver` implementation decides `_chip_dnssd_default_platform` and `_chip_dnssd_default_minimal` flags that cannot be enabled at the same time.

In short we can compile both solutions, but only one can be set as a default one. The other one still can be set dynamically.